### PR TITLE
Expand OtlpMetricExporter tests

### PR DESCRIPTION
**Pull request by alanwest**
Fixes #.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs [hunk 1]
```diff
@@ -49,99 +46,326 @@ public void ToOtlpResourceMetricsTest(bool includeServiceNameInResource)
...
+            var metrics = new List<Metric>();
+
+            using var meter = new Meter(Utils.GetCurrentMethodName());
+            using var provider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddInMemoryExporter(metrics)
+                .Build();
 
-            var testCompleted = false;
+            var attributes = ToAttributes(keysValues).ToArray();
```
Comment (cijothomas) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs:103]: doesnt seem used.

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs [hunk 2]
```diff
@@ -49,99 +46,326 @@ public void ToOtlpResourceMetricsTest(bool includeServiceNameInResource)
...
+        [InlineData("test_counter", null, null, 123, null, AggregationTemporality.Cumulative, true)]
+        [InlineData("test_counter", null, null, null, 123.45, AggregationTemporality.Cumulative, true)]
+        [InlineData("test_counter", null, null, 123, null, AggregationTemporality.Delta, true)]
+        [InlineData("test_counter", "description", "unit", 123, null, AggregationTemporality.Cumulative, true)]
+        [InlineData("test_counter", null, null, 123, null, AggregationTemporality.Delta, true, "key1", "value1", "key2", 123)]
+        public void TestCounterToOltpMetric(string name, string description, string unit, long? longValue, double? doubleValue, AggregationTemporality aggregationTemporality, bool isMonotonic, params object[] keysValues)
+        {
+            var metrics = new List<Metric>();
+
+            var metricReader = new BaseExportingMetricReader(new InMemoryExporter<Metric>(metrics));
```
Comment (reyang) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs:181]: Minor: consider this if we don't need the metricReader local variable https://github.com/open-telemetry/opentelemetry-dotnet/blob/fb161af9393f1eb2a6bae077f2478700a1623632/test/OpenTelemetry.Tests/Metrics/MemoryEfficiencyTests.cs#L38-L42

**Review by reyang — APPROVED** [review]
LGTM.

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs (hunk 1 above)
Comment (alanwest) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs]: Yep you're right. I removed it . No way to set attributes using ObservableGauge right?

Comment (cijothomas) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs]: Observable can also have attributes! See this example
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/metrics/learning-more-instruments/Program.cs#L56

Comment (reyang) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs]: https://github.com/open-telemetry/opentelemetry-dotnet/blob/794450b17808738a9f66543691a8cfaab1d83d9b/examples/Console/TestPrometheusExporter.cs#L63

**Pull request merged**